usvm-ts/src/main/kotlin/org/usvm/machine/expr/TsLocalResolver.kt

```diff
@@ -0,0 +1,199 @@
+            scope.doWithState {
+                initializeGlobals(file)
+            }
+            return null
+        } else {
+            // TODO: handle methodResult
+            scope.doWithState {
+                if (methodResult is TsMethodResult.Success) {
+                    methodResult = TsMethodResult.NoCall
+                }
+            }
+        }
+
+        // Try to get the saved sort for this dflt object field
+        val savedSort = scope.calcOnState {
+            getSortForDfltObjectField(file, local.name)
+        }
+
+        if (savedSort == null) {
+            // No saved sort means this field was never assigned to, which is an error
+            logger.error { "Trying to read unassigned global variable: '$local' in $file" }
+            scope.assert(falseExpr)
+            return null
+        }
+
+        // Use the saved sort to read the field
+        return mkFieldLValue(savedSort, dfltObject, local.name)
+    }
+
+    // If local is an imported variable:
+    val importInfo = file.importInfos.find { it.name == local.name }
+    if (importInfo != null) {
+        return when (val resolutionResult = scene.resolveImportInfo(file, importInfo)) {
+            is SymbolResolutionResult.Success -> {
+                val importedFile = resolutionResult.file
+                val importedDfltObject = scope.calcOnState { getDfltObject(importedFile) }
+
+                // Initialize globals in the imported file if necessary
+                val isImportedFileGlobalsInitialized = scope.calcOnState { isGlobalsInitialized(importedFile) }
+                if (!isImportedFileGlobalsInitialized) {
+                    logger.info { "Globals are not initialized for imported file: $importedFile" }
+                    scope.doWithState {
+                        initializeGlobals(importedFile)
+                    }
+                    return null
+                }
+
+                if (resolutionResult.exportInfo.name == "*") {
+                    logger.warn { "Star import" }
+                    val lValue = mkFieldLValue(addressSort, importedDfltObject, "__self__")
+                    scope.doWithState {
+                        memory.write(lValue, importedDfltObject, guard = trueExpr)
+                    }
+                    return lValue
+                }
+
+                // Try to get the saved sort for this imported dflt object field
+                val symbolNameInImportedFile = resolutionResult.exportInfo.originalName
+                val savedSort = scope.calcOnState {
+                    getSortForDfltObjectField(importedFile, symbolNameInImportedFile)
+                }
+
+                if (savedSort == null) {
+                    // No saved sort means this field was never assigned to, which is an error
+                    logger.error { "Trying to read unassigned imported symbol: '$local' from '$importedFile'" }
+                    scope.assert(falseExpr)
```

Comment:
If this is an error, you should throw an error. Not simply remove the state that will be an issue for someone later